### PR TITLE
Add a global exception handler (CI issue solving)

### PR DESCRIPTION
**Pull request by sinkingsugar**
Should be commented by default
It works great, check the logs.. shows the whole path an exception took form begin to end.
@dryajov @arnetheduck

### CODE REVIEW

**Comment by sinkingsugar** [comment]
We have a lot of `name=CancelledError`, but not sure we are doing proper propagation indeed.

**Comment by sinkingsugar** [comment]
I've re run the PR multiple times, I have to conclude that indeed the excess of logging was causing something in travis. In fact only travis CI failed.

File: libp2p/errors.nim [hunk 1]
```diff
@@ -5,6 +5,14 @@ import chronos
 import chronicles
 import macros
 
+# Uncomment only in dire need
+globalRaiseHook = proc (e: ref Exception): bool {.gcsafe, locks: 0.} =
```
Comment (arnetheduck) [inline libp2p/errors.nim:9]: the usual way for these kinds of debugging helps is through a `-d` flag

Comment (sinkingsugar) [reply libp2p/errors.nim:9]: Sure, but I'm not even sure I want to push anything from this PR as I don't think it's something that needs to be in the repo but rather do it while developing. As I often do.

This PR is special cos I could not reproduce locally. Had to deploy and always wanted to share the logs.

Comment (sinkingsugar) [reply libp2p/errors.nim:9]: Do you guys actually think this helper should be in the code base behind something like `debug_exceptions` or so?

Comment (dryajov) [reply libp2p/errors.nim:9]: Pushing is fine as long as it's behind a flag.

`debug_exceptions` is as good as any, or what I've been leaning towards is piggybacking from the chronicles log levels to some extent, for example I'm using trace level to enable most other debugging facilities info. Either way we don't and shouldn't for the most part log anything too verbose unless tracing.

**Comment by sinkingsugar** [comment]
Closing this for now, anyway was just a proof of concept/idea to debug

**Pull request closed without merging**